From 297f635bfffca114ae9ba172860e91b22d3b23da Mon Sep 17 00:00:00 2001
From: Masahiro Yamada <masahiroy@kernel.org>
Date: Sat, 22 Jul 2023 13:48:00 +0900
Subject: [PATCH 17/58] kbuild: rpm-pkg: run modules_install for non-modular
 kernel

For the same reason as commit 4243afdb9326 ("kbuild: builddeb: always
make modules_install, to install modules.builtin*"), run modules_install
even when CONFIG_MODULES=n to install modules.builtin*.

Signed-off-by: Masahiro Yamada <masahiroy@kernel.org>
---
 scripts/package/mkspec | 15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

diff --git a/scripts/package/mkspec b/scripts/package/mkspec
index a9425d993..2613e85cd 100755
--- a/scripts/package/mkspec
+++ b/scripts/package/mkspec
@@ -101,16 +101,13 @@ $S$M
 	%else
 	cp \$(%{make} %{makeflags} -s image_name) %{buildroot}/boot/vmlinuz-%{KERNELRELEASE}
 	%endif
-$M	%{make} %{makeflags} INSTALL_MOD_PATH=%{buildroot} modules_install
+	%{make} %{makeflags} INSTALL_MOD_PATH=%{buildroot} modules_install
 	%{make} %{makeflags} INSTALL_HDR_PATH=%{buildroot}/usr headers_install
 	cp System.map %{buildroot}/boot/System.map-%{KERNELRELEASE}
 	cp .config %{buildroot}/boot/config-%{KERNELRELEASE}
-$S$M	rm -f %{buildroot}/lib/modules/%{KERNELRELEASE}/build
-$S$M	rm -f %{buildroot}/lib/modules/%{KERNELRELEASE}/source
+	ln -fns /usr/src/kernels/%{KERNELRELEASE} %{buildroot}/lib/modules/%{KERNELRELEASE}/build
+	ln -fns /usr/src/kernels/%{KERNELRELEASE} %{buildroot}/lib/modules/%{KERNELRELEASE}/source
 $S$M	%{make} %{makeflags} run-command KBUILD_RUN_COMMAND='\${srctree}/scripts/package/install-extmod-build %{buildroot}/usr/src/kernels/%{KERNELRELEASE}'
-$S$M	cd %{buildroot}/lib/modules/%{KERNELRELEASE}
-$S$M	ln -sf /usr/src/kernels/%{KERNELRELEASE} build
-$S$M	ln -sf /usr/src/kernels/%{KERNELRELEASE} source
 
 	%clean
 	rm -rf %{buildroot}
@@ -138,9 +135,9 @@ $S$M	ln -sf /usr/src/kernels/%{KERNELRELEASE} source
 
 	%files
 	%defattr (-, root, root)
-$M	/lib/modules/%{KERNELRELEASE}
-$M	%exclude /lib/modules/%{KERNELRELEASE}/build
-$M	%exclude /lib/modules/%{KERNELRELEASE}/source
+	/lib/modules/%{KERNELRELEASE}
+	%exclude /lib/modules/%{KERNELRELEASE}/build
+	%exclude /lib/modules/%{KERNELRELEASE}/source
 	/boot/*
 
 	%files headers
-- 
2.42.0

